langsmith-java-core/src/main/java/com/langchain/smith/wrappers/openai/OpenTelemetryConfig.java

```
@@ -65,6 +71,234 @@ public enum SpanProcessorType {
+         */
+        public Builder processorType(SpanProcessorType processorType) {
+            this.processorType = processorType;
+            return this;
+        }
+
+        /**
+         * Sets the maximum batch size for span export (optional).
+         *
+         * <p>
+         * <b>Only applies to BATCH processor type.</b> Ignored when using SIMPLE processor.
+         *
+         * <p>
+         * Controls how many spans are batched together before triggering an export:
+         * <ul>
+         * <li><b>Default: 512</b> - Good balance between efficiency and latency</li>
+         * <li><b>1</b> - Exports immediately (still non-blocking unlike SIMPLE)</li>
+         * <li><b>Higher values</b> - More efficient but increased latency</li>
+         * </ul>
+         *
+         * <p>
+         * <b>Setting to 1:</b> If you want immediate export but don't want to block
+         * your application thread, use BATCH processor with maxBatchSize=1 instead of
+         * SIMPLE processor. This gives you non-blocking immediate export.
+         *
+         * <p>
+         * Example for immediate non-blocking export:
+         * <pre>{@code
+         * OpenTelemetryConfig.builder()
+         *     .apiKey(apiKey)
+         *     .processorType(SpanProcessorType.BATCH)
+         *     .maxBatchSize(1)  // Export immediately but don't block
+         *     .build();
+         * }</pre>
+         *
+         * @param maxBatchSize the maximum batch size (1-512, default 512)
+         * @return this Builder for method chaining
+         */
+        public Builder maxBatchSize(int maxBatchSize) {
+            this.maxBatchSize = maxBatchSize;
+            return this;
+        }
+
+        /**
+         * Builds and configures the OpenTelemetry instance.
+         *
+         * @return the configured OpenTelemetry instance
+         * @throws IllegalArgumentException if apiKey is not set
+         */
+        public OpenTelemetry build() {
```

probably want to do some validation here and throw up if the parameters are missing, instead of getting an NPE down the line.

---

yeah I had a ticket to refactor that but done it here